xla/stream_executor/rocm/rocm_executor.cc

```diff
@@ -455,6 +455,20 @@ void GpuExecutor::Deallocate(DeviceMemoryBase* mem) {
   GpuDriver::DeviceDeallocate(context_, mem->opaque());
 }
 
+absl::StatusOr<std::unique_ptr<MemoryAllocation>>
+GpuExecutor::HostMemoryAllocate(uint64_t size) {
+  auto* buffer = GpuDriver::HostAllocate(context_, size);
+  if (buffer == nullptr && size > 0) {
+    return absl::InternalError(
+        absl::StrFormat("Failed to allocate HostMemory of size %d", size));
+  }
+  return std::make_unique<HostMemoryAllocation>(buffer, size, this);
+}
+
+void GpuExecutor::HostMemoryDeallocate(void* location, uint64_t size) {
```

Comment:
I pushed a change that tries to smooth this over by keeping the two signatures in parallel. I am not sure how to test the impact in TensorFlow.

This will not allow all combinations to work, as I think that could only be achieved by adding an extra runtime data structure tracking {ptr: size}, but it should allow old implementations of the old interface to continue working.

Reply:
I will attempt to fix up this CL internally, and try and get it submitted today.